src/libsyntax/feature_gate.rs

```diff
@@ -1105,12 +1108,28 @@ impl<'a> Context<'a> {
             // before the plugin attributes are registered
             // so we skip this then
             if !is_macro {
-                gate_feature!(self, custom_attribute, attr.span,
-                              &format!("The attribute `{}` is currently \
-                                        unknown to the compiler and \
-                                        may have meaning \
-                                        added to it in the future",
-                                       attr.path));
+                if attr.is_scoped() {
+                    gate_feature!(self, tool_attributes, attr.span,
```

Comment:
This solution is 100% technical debt, but I guess it can work while everything is unstable, and we go into this branch only if previous ways to resolve the attribute failed.

I described the proper solution in https://github.com/rust-lang/rfcs/pull/2103#discussion_r136705802.
TLDR: `a::b::c` in `#[a::b::c(...)]` should be resolved as a usual path in macro namespace and therefore `rustfmt` and `clippy` should be added into the [language prelude](https://github.com/rust-lang/rust/blob/master/src/librustc_resolve/lib.rs#L1204-L1220) as special modules (primitive types `u8`, `u16`, etc work in very similar way).
I don't suggest to implement this now though.

Reply:
>I don't suggest to implement this now though.

But I suggest to implement this *eventually*, if you plan to continue working on macros/attributes and name resolution in rustc.